edge-hub/src/Microsoft.Azure.Devices.Routing.Core/Route.cs

```diff
@@ -11,13 +11,15 @@ namespace Microsoft.Azure.Devices.Routing.Core
 
     public class Route : IEquatable<Route>
     {
-        public Route(string id, string condition, string iotHubName, IMessageSource source, ISet<Endpoint> endpoints)
```

Comment:
Just curious, I found the usages of this constructor in RouteFactory; and it always pass only 1 element in endpoints argument.  Do you know why this argument type is ISet<Endpoint>? what use cases?

Reply:
I talked to Varun about this, it's legacy code from when we adapted the routing stack from IoT Hub.  We don't actually have a usecase that takes more than 1 endpoint, that's why I cleaned it up.  It would have actually taken more time if I needed to update every caller with the older signature.

---
In reply to: [377445441](https://github.com/Azure/iotedge/pull/2469#discussion_r377445441) [](ancestors = 377445441)